engines/bops_core/app/controllers/concerns/bops_core/application_controller.rb

```diff
@@ -64,12 +64,24 @@ def current_api_user
     end
 
     def authenticate_api_user!
```

Comment:
Feel like this logic could be broken down into a concern so the controller remains focused on handling HTTP requests while the concern handles finding the ApiUser and enforcing permissions. But perhaps a wider observation than just here.